### PR TITLE
[test] Add PrgEnv-nvidia to the mpi, tools and ulimit tests

### PR DESCRIPTION
**Pull request by jgphpc**
epic: https://jira.cscs.ch/browse/UES-1630

### CODE REVIEW

File: cscs-checks/tools/profiling_and_debugging/notool.py [hunk 1]
```diff
@@ -75,16 +68,51 @@ def __init__(self):
...
             f'echo CRAY_AOCC_VERSION=$CRAY_AOCC_VERSION',
         ]
         self.reference = {'*': {'elapsed_time': (0, None, None, 's')}}
-        if self.lang == 'C++':
-            self.reference_lang = (0.38, -0.6, None, 's')
-        elif self.lang == 'F90':
-            self.reference_lang = (0.17, -0.6, None, 's')
+
+    @run_before('compile')
+    def deal_with_prgEnv_nvidia(self):
```
Comment (jjotero) [inline cscs-checks/tools/profiling_and_debugging/notool.py:77]:
```suggestion
    def prgEnv_nvidia_workaround(self):
```

**Review by jjotero — COMMENTED** [review]
My suggestions are mostly about moving as many things as possible into the class body since this improves the readability of the test quite a lot. But other than that, these changes look good.

File: cscs-checks/tools/profiling_and_debugging/notool.py [hunk 2]
```diff
@@ -75,16 +68,52 @@ def __init__(self):
...
+        # FIXME: workaround for C4KCUST-308
+        self.modules += ['cray-mpich']
+        self.prgenv_flags = {
+            'PrgEnv-aocc': ['-O2', '-g', '-fopenmp'],
+            'PrgEnv-cray': ['-O2', '-g',
+                            '-homp' if self.lang == 'F90' else '-fopenmp'],
+            'PrgEnv-gnu': ['-O2', '-g', '-fopenmp'],
+            'PrgEnv-intel': ['-O2', '-g', '-qopenmp'],
+            'PrgEnv-pgi': ['-O2', '-g', '-mp'],
+            'PrgEnv-nvidia': ['-O2', '-g', '-mp', self.cppflags]
```
Comment (jjotero) [inline cscs-checks/tools/profiling_and_debugging/notool.py:115]: `cppflags` here is `None` and that causes an error.

File: cscs-checks/tools/profiling_and_debugging/notool.py [hunk 3]
```diff
@@ -75,16 +68,52 @@ def __init__(self):
...
+        # cdt/20.08 cray-mpich/7.7.15 cray, crayclang, gnu, intel, pgi
+        # cdt/21.02 cray-mpich/7.7.16 cray, crayclang, gnu, intel, pgi
+        # cdt/21.05 cray-mpich/7.7.17 crayclang, gnu, intel, pgi, *nvidia*
+        #
+        # cpe/21.04 cray-mpich/8.1.4 AOCC, CRAY, CRAYCLANG, GNU, INTEL, NVIDIA
+        # cpe/21.05 cray-mpich/8.1.5 AOCC, CRAY, CRAYCLANG, GNU, INTEL, NVIDIA
+        # cpe/21.06 cray-mpich/8.1.6 AOCC, CRAY, CRAYCLANG, GNU, INTEL, NVIDIA
+        # }}}
+        envname = self.current_environ.name
+        self.cppflags = None
```
Comment (jjotero) [inline cscs-checks/tools/profiling_and_debugging/notool.py:88]: I think this could solve it.
```suggestion
        self.cppflags = ''
```

**Comment by jgphpc** [comment]
retest this please

> pytest.fail(f'process {pid} is still alive')

**Comment by jgphpc** [comment]
retest this please

>       assert osext.git_repo_exists

**Comment by jgphpc** [comment]
The check has passed, the unittest  fails in:

**Review by jjotero — APPROVED** [review]

**Comment by vkarak** [comment]
@jenkins-cscs retry eiger

**Pull request merged**